cmd/goal/clerk.go

```diff
@@ -1070,7 +1070,7 @@ var compileCmd = &cobra.Command{
 				}
 			}
 			if writeSourceMap {
-				mapname := fname + ".map"
+				mapname := outname + ".map"
```

Comment:
What if `outname` is `stdoutFilenameValue`, which is "-"? I think `fmt.Sprintf("%s.tok.map", fname)` would be a better choice in that case

Reply:
Good catch, I'll change this - what is the purpose of having a `stdoutFilenameValue`?

Reply:
Oh wait my suggestion is broken for this reason: if `outname` is `stdoutFilenameValue`, then `fname` _must_ be `stdinFileNameValue`, which is also "-".

Reply:
I guess the only thing you can do is error?

The situation would only happen if someone is piping into goal and out of it while also specifying maps should be generated, e.g. `cat ... > goal clerk compile -m > some_other_command`. It appears to me there's no intuitive place for the maps to go.

Reply:
Ah okay, I see why it's `-` now (literally for stdin/stdout). I'll error out in this case for now (but I wouldn't be opposed to printing it to stdout either).